application/src/main/java/org/togetherjava/tjbot/features/jshell/renderer/ResultGistRenderer.java

```diff
@@ -0,0 +1,142 @@
+        builder.append("// ").append(getGeneralStatus(result)).append("\n");
+
+        if (originator != null) {
+            builder.append("// ").append(originator.getEffectiveName()).append("'s result\n\n");
+        }
+
+        setResultToBuilder(showCode, builder, result);
+
+        String text = builder.toString();
+
+        GHGist gist;
+        try {
+            GHGistBuilder gistBuilder = new GitHubBuilder().withOAuthToken(gistApiToken)
+                .build()
+                .createGist()
+                .public_(false);
+            if (originator != null) {
+                gistBuilder.description("Uploaded by " + originator.getEffectiveName());
+            }
+            gistBuilder.file((originator == null ? "JShell" : originator.getEffectiveName())
+                    + "'s result.java", text);
+
+            gist = gistBuilder.create();
+        } catch (IOException ex) {
+            logger.error("Couldn't send JShell result to Gist", ex);
+            return Optional.empty();
+        }
+
+        EmbedBuilder embedBuilder = new EmbedBuilder()
+            .setTitle("The result was too big and so was uploaded to Gist: " + gist.getHtmlUrl()
+                    + "\n↓This is the output of the JShell execution.↓")
+            .setDescription(getSdtOut(result).transform(s -> s.isBlank() ? "Nothing" : s))
+            .setColor(RendererUtils.getStatusColor(result));
+        if (originator != null) {
+            embedBuilder.setAuthor(originator.getEffectiveName() + "'s result");
+        }
+        return Optional.of(embedBuilder.build());
+    }
+
+    private void setResultToBuilder(boolean showCode, StringBuilder builder, JShellResult result) {
+        if (showCode) {
+            for (JShellSnippetResult r : result.snippetsResults()) {
+                builder.append("""
+                        // Snippet %d, %s
+                        %s
+                        jshell> %s
+                        """.formatted(r.id(), r.status(), r.source(),
+                        r.result() == null ? "" : r.result()));
+            }
+        }
+        if (result.abortion() != null) {
+            builder.append('\n');
+            builder.append("// [WARNING] The code couldn't end properly...\n");
+            builder.append(abortionToString(result.abortion()));
+        }
+        setStdoutToBuilder(builder, result);
+    }
+
+    private void setStdoutToBuilder(StringBuilder builder, JShellResult result) {
+        if (result.stdout().isEmpty()) {
+            builder.append("// No result");
+        } else {
+            builder.append("// Result:\n");
+            builder.append(getSdtOutAsComment(result));
+        }
+    }
+
+    private String getSdtOut(JShellResult result) {
+        String stdout = result.stdout().replace("\n", "\n// ");
+        if (result.stdoutOverflow()) {
+            stdout += MessageUtils.ABBREVIATION;
+        }
+        return stdout;
+    }
```

Comment:
Duplicate method, check RendererUtils. Maybe check other methods as well, might be more duplicates

Reply:
Oh it's not the same?
ok mb didn't look closely ig